redis_cluster_handler.go

```diff
@@ -128,13 +128,25 @@ func (r *RedisClusterStorageManager) cleanKey(keyName string) string {
 	return setKeyName
 }
 
-// GetKey will retrieve a key from the database
-func (r *RedisClusterStorageManager) GetKey(keyName string) (string, error) {
-	if GetRelevantClusterReference(r.IsCache) == nil {
-		log.Info("Connection dropped, connecting..")
+func (r *RedisClusterStorageManager) ensureConnection() {
+	if GetRelevantClusterReference(r.IsCache) != nil {
+		// already connected
+		return
+	}
+	log.Info("Connection dropped, reconnecting...")
+	for {
```

Comment:
These two things are what I was talking about on slack and in the original issue.

This PR is not about changing or fixing the logic, it's about deduplicating it. I simply rewrote the recursive call into a for loop. Which makes the broken logic more obvious too.